### PR TITLE
Move field generator to libbeat and use it from the beats

### PR DESCRIPTION
**Pull request by ruflin**
See https://github.com/elastic/beats/issues/807

### CODE REVIEW

**Comment by ruflin** [comment]
@monicasarbu @dedemorton Can you have a look at this?

File: filebeat/scripts/field_docs.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    ("log", "Log File Content"),
+]
+
+
+if __name__ == "__main__":
+    if len(sys.argv) != 3:
+        print "Usage: %s file.yml file.asciidoc" % sys.argv[0]
+        sys.exit(1)
+
+    input = open(sys.argv[1], 'r')
```
Comment (urso) [inline filebeat/scripts/field_docs.py]: with open(...) as input:
        with open(...) as output:
            ...

File: filebeat/scripts/field_docs.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    output = open(sys.argv[2], 'w')
+
+    try:
+        generate_fields_docs.fields_to_asciidoc(input, output, SECTIONS, "Filebeat")
```
Comment (urso) [inline filebeat/scripts/field_docs.py]: can fields_to_asciidoc check if input/output is a string and open files. This 30 something lines boilerplate still looks like too much boilerplate.

Comment (ruflin) [inline filebeat/scripts/field_docs.py]: The code above is just copy / paste from the existing code. I plan to do some refactoring in a second step.

**Comment by monicasarbu** [comment]
LGTM. Just a small thing. The generated doc doesn't look properly. We need to adjust the indentation.
screen shot 2016-02-01 at 7 13 36 pm

**Pull request merged**